adapter/authzOpa/authzOpa.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authzOpa // import "istio.io/mixer/adapter/authzOpa"
+
+// NOTE: This adapter will eventually be auto-generated so that it automatically supports all CHECK and QUOTA
+//       templates known to Mixer. For now, it's manually curated.
+
+import (
+	"context"
+	"fmt"
+
+	rpc "github.com/googleapis/googleapis/google/rpc"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+
+	"istio.io/mixer/adapter/authzOpa/config"
+	"istio.io/mixer/pkg/adapter"
+	"istio.io/mixer/pkg/status"
+	"istio.io/mixer/template/authz"
+)
+
+type (
+	builder struct {
+		adapterConfig *config.Params
+	}
+
+	handler struct {
+		policy      string
+		checkMethod string
+
+		context  context.Context
+		compiler *ast.Compiler
+
+		env adapter.Env
+	}
+)
+
+///////////////// Configuration Methods ///////////////
+
+func (b *builder) SetAuthzTypes(map[string]*authz.Type) {}
+
+func (b *builder) SetAdapterConfig(cfg adapter.Config) {
+	b.adapterConfig = cfg.(*config.Params)
+}
+
+func (b *builder) Validate() (ce *adapter.ConfigErrors) {
+	if len(b.adapterConfig.CheckMethod) == 0 {
+		ce = ce.Appendf(GetInfo().Name,
+			"CheckMethod was not configured")
+	}
+
+	parsed, err := ast.ParseModule("", string(b.adapterConfig.Policy))
+	if err != nil {
+		ce = ce.Appendf(GetInfo().Name,
+			"Failed to parse the OPA policy: %v", err)
+	}
+
+	compiler := ast.NewCompiler()
+	compiler.Compile(map[string]*ast.Module{"": parsed})
+	if compiler.Failed() {
+		ce = ce.Appendf(GetInfo().Name,
+			"Failed to compile the OPA policy: %v", compiler.Errors)
+	}
+
+	return
+}
+
+func (b *builder) Build(context context.Context, env adapter.Env) (adapter.Handler, error) {
+	h := handler{
+		policy:      b.adapterConfig.Policy,
+		checkMethod: b.adapterConfig.CheckMethod,
+		env:         env,
+	}
+
+	parsed, _ := ast.ParseModule("", string(h.policy))
+
+	compiler := ast.NewCompiler()
+	compiler.Compile(map[string]*ast.Module{"": parsed})
+
+	h.compiler = compiler
+	h.context = context
```

Comment:
I don't think you need to hold on to this context: to intent is for it to be used in any RPCs made during Build as part of setup, to enable cancelation of the build and that kind of thing (no existing adapter uses it for this today, but it's a brand new part of the interface).

Reply:
Removed